intersect.go

```diff
@@ -176,6 +176,17 @@ func Without[T comparable, Slice ~[]T](collection Slice, exclude ...T) Slice {
 	return result
 }
 
+// WithoutBy returns a slice excluding values whose extracted key is in the exclude list.
```

Comment:
This is can be a much better comment I think:

// WithoutBy filters a slice by excluding elements whose extracted keys match any in the exclude list.
// It returns a new slice containing only the elements whose keys are not in the exclude list.

Reply:
Thanks, I modifed this comments. Please review it again.